include/deal.II/base/tensor_base.h

```diff
@@ -854,28 +836,59 @@ void Tensor<0,dim,Number>::serialize(Archive &ar, const unsigned int)
   ar &value;
 }
 
-/*---------------------------- Inline functions: Tensor<1,dim,Number> ------------------------*/
 
+/*-------------------- Inline functions: Tensor<rank,dim> --------------------*/
 
-template <int dim, typename Number>
+
+namespace internal
+{
+  // TODO: Think about refactoring this into the TableIndices class as a
+  // general, polymorphic for extracting an item out of an object with
+  // nested identifiers.
+  template<int rank_> struct TensorIndicesHelper
+  {
+    // used for implementing Tensor<rank,dim>::operator[] with TableIndices
+    // tail recursive call to form up access to
+    //   tensor[indices[0]][indices[1]]...[indices[rank_]]
+    template<int rank, int dim, typename Number>
+    static inline
+    Number &extract(Tensor<rank_,dim,Number> &t, const TableIndices<rank> &indices)
```

Comment:
In principle yes. Except: Due to type ambiguity I have to specify the recursive template parameters. I cannot do this with function parameter overloading.